inst/rmarkdown/templates/oup/resources/template.tex

```diff
@@ -96,48 +122,76 @@
   \fi
 }
 
-\newcommand*\thanksbycode[1]{%
-  \ifcounter{FNCT@#1}
-    {\samethanks[\value{FNCT@#1}]}
-    {\thanks{\Ref{FN}{#1}}\newcounter{FNCT@#1}\setcounter{FNCT@#1}{\value{footnote}}}
+\newcommand*\addrlblbycode[1]{%
+  \ifcounter{ADDRLBL@#1}
+    {}
+    {\refstepcounter{addrcnt}\newcounter{ADDRLBL@#1}\setcounter{ADDRLBL@#1}{\value{addrcnt}}}%
+    \arabic{ADDRLBL@#1}%
+}
+
+\newcommand*\addrbycode[1]{%
+  \ifcounter{ADDR@#1}
+    {}
+    {\newcounter{ADDR@#1}%
+     \address[\addrlblbycode{#1}]{\MyRef{ADDRTXT}{#1}}}%
+}
+
+\newcommand*\corresplblbycode[1]{%
+  \ifcounter{CORRESPLBL@#1}
+    {}
+    {\refstepcounter{correspcnt}\newcounter{CORRESPLBL@#1}\setcounter{CORRESPLBL@#1}{\value{correspcnt}}}%
+    \fnsymbol{CORRESPLBL@#1}%
+}
+
+\newcommand*\correspbycode[1]{%
+  \ifcounter{CORRESP@#1}
+    {}
+    {\newcounter{CORRESP@#1}%
+     \corresp[\corresplblbycode{#1}]{\MyRef{CORRESPTXT}{#1}}}%
 }
 
+\makeatother
+
+% Add missing \city command mentioned in documentation but absent from cls
+\providecommand\city[1]{#1}
+
 % Create labels for Addresses if the are given in Elsevier format
-$if(address)$
 $for(address)$
-$if(address.code)$
-\Newlabel{ADR}{$address.code$}{$address.address$}
-$endif$
-$if(address.id)$
-\Newlabel{ADR}{$address.id$}{$address.address$}
-$endif$
+ $if(address.code)$
+  \MyNewLabel{ADDRTXT}{$address.code$}{%
+ $elseif(address.id)$
+  \MyNewLabel{ADDRTXT}{$address.id$}{%
+ $endif$
+ $if(address.address)$$address.address$$else$%
+  $if(address.department)$\orgdiv{$address.department$}, $endif$%
+  $if(address.organization)$\orgname{$address.organization$}, $endif$%
+  $if(address.street)$\orgaddress{%
+   $elseif(address.city)$\orgaddress{%
+   $elseif(address.postcode)$\orgaddress{%
+   $elseif(address.state)$\orgaddress{%
+   $elseif(address.country)$\orgaddress{%
+  $endif$%
+  $if(address.street)$\street{$address.street$}, $endif$%
+  $if(address.postcode)$\postcode{$address.postcode$}, $endif$%
+  $if(address.city)$\city{$address.city$}, $endif$%
+  $if(address.state)$\state{$address.state$}, $endif$%
+  $if(address.country)$\country{$address.country$}$endif$%
+  $if(address.street)$}%
+   $elseif(address.city)$}%
+   $elseif(address.postcode)$}%
+   $elseif(address.state)$}%
+   $elseif(address.country)$}%
+  $endif$%
+ $endif$%
+ }
```

Comment:
This is a breaking change for previous user if they rerun their article using `oup_article()` ?

Reply:
I have tried to make this not be a breaking change for articles written using the old format. The code you highlight has two functions:

1) The first part associates codes with addresses. This is necessary for dealing with multiple authors with the same address.
2) Format addresses that are given in the new format that is split into a bunch of pieces.

If users used the older `$address.address$`, then it should still work.

Reply:
Ok thanks !